src/main/scala/io/getclump/ClumpFetcher.scala

```diff
@@ -1,10 +1,10 @@
 package io.getclump
 
-import scala.collection.mutable.{Map => MutableMap}
+import scala.collection.mutable
 
 private[getclump] final class ClumpFetcher[T, U](source: ClumpSource[T, U]) {
 
-  private[this] val fetches = MutableMap[T, Promise[Option[U]]]()
+  private[this] val fetches = mutable.LinkedHashMap[T, Promise[Option[U]]]()
```

Comment:
Because it retains insertion order. From [the docs](http://www.scala-lang.org/api/2.11.5/index.html#scala.collection.mutable.LinkedHashMap):

`The iterator and all traversal methods of this class visit elements in the order they were inserted.`